src/sidebar/filter/sidebar-tutorial.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+/**
+ * Returns true if the sidebar tutorial has to be shown to a user for a given session.
+ */
+function show(sessionState) {
```

Comment:
- The `preferences` property is a non-optional field in the profile type as returned by `/api/profile`, so you don't need to check for it.
- Re. your question about where this should go, the `src/sidebar/filter` directory refers to Angular filters, rather than utility functions. There are two places in the code for this kind of logic:

  1. A simple function in a module in the `util/` directory (eg. in say `util/session` or something). If you do this, please make sure the function name makes sense on its own (eg. `shouldShowSidebarTutorial`).
  2. A "selector" function in an app state module (eg. `src/sidebar/reducers/session` in this case) which takes the app state and returns something derived from it.

Also, one related note - the name "session" in various modules is a hangover from when the data came from h's `/app` endpoint, which returned data about the user's cookie session on the website. At some point we should change it to "profile" everywhere since it comes from `/api/profile` now and no longer necessarily reflects the user's state on the website (eg. if using third party accounts).